native_client/javascript/client.ts

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env node
+
+import Ds from "deepspeech";
```

Comment:
@piraka9011 Our current packaging of `deepspeech`, `deepspeech-tflite` and `deepspeech-gpu` does not work with this statement.

Currently, `client.js` loads `./index.js` that performs `node-pre-gyp` loading.

Trying to use `import Ds from "./index";` fails with:
```
$ PATH=$HOME/node_modules/.bin/:$PATH tsc
client.ts:3:16 - error TS2306: File '/home/alex/codaz/Mozilla/DeepSpeech/deepspeech-kdavis/native_client/javascript/index.d.ts' is not a module.

3 import Ds from "./index";
                 ~~~~~~~~~


Found 1 error.
```

Which is not super-intuitive to debug, especially since `index.d.ts` does declare a module:
```
$ head index.d.ts 
declare module 'deepspeech' {
/**
 * Stores text of an individual token, along with its timing information
 */
export interface TokenMetadata {
    text: string;
    timestep: number;
    start_time: number;
}
```

Given the lack of debuggability of `tsc`, I could only verify that `index.d.ts` is indeed the loaded file.
Now, my understand is that it should rather be `./index.js`, but that is impossible because it also fails on `index.d.ts` not being a module.

If I rename our `index.js` to `lib.js`, then:
```
client.ts:3:16 - error TS7016: Could not find a declaration file for module './lib.js'. '/home/alex/codaz/Mozilla/DeepSpeech/deepspeech-kdavis/native_client/javascript/lib.js' implicitly has an 'any' type.
```

Reply:
I can workaround that by removing the `declare module "deepspeech" {}` lines.

Reply:
There is no `Ds` Class or method in the `deepspeech` module (specifically the type declaration file. If you want to use it, it should either be:

```
import * as ds from 'deepspeech'
```
of
```
// Or whatever interface/class
import {Model} from 'deepspeech'
import {Metadata} from 'deepspeech'
```

What you did I guess is technically correct... Though I'm not sure how it would work when used outside the package.

Reply:
> `import * as ds from 'deepspeech'`

I tried that, but it was not working either as much as I can tell



> What you did I guess is technically correct... Though I'm not sure how it would work when used outside the package.

I could successfully `tsc` a small `test.ts` after installing my `npm` package, so I guess we are safe ?

Reply:
```
alex@portable-alex:~/codaz/Mozilla/DeepSpeech$ head -n 15 ds.log 
======== Resolving module 'deepspeech-tflite' from '/home/alex/codaz/Mozilla/DeepSpeech/test.ts'. ========
Module resolution kind is not specified, using 'NodeJs'.
Loading module 'deepspeech-tflite' from 'node_modules' folder, target file type 'TypeScript'.
Directory '/home/alex/codaz/Mozilla/DeepSpeech/node_modules' does not exist, skipping all lookups in it.
Directory '/home/alex/codaz/Mozilla/node_modules' does not exist, skipping all lookups in it.
Directory '/home/alex/codaz/node_modules' does not exist, skipping all lookups in it.
Found 'package.json' at '/home/alex/node_modules/deepspeech-tflite/package.json'.
'package.json' does not have a 'typesVersions' field.
File '/home/alex/node_modules/deepspeech-tflite.ts' does not exist.
File '/home/alex/node_modules/deepspeech-tflite.tsx' does not exist.
File '/home/alex/node_modules/deepspeech-tflite.d.ts' does not exist.
'package.json' does not have a 'typings' field.
'package.json' has 'types' field './index.d.ts' that references '/home/alex/node_modules/deepspeech-tflite/index.d.ts'.
File '/home/alex/node_modules/deepspeech-tflite/index.d.ts' exist - use it as a name resolution result.
Resolving real path for '/home/alex/node_modules/deepspeech-tflite/index.d.ts', result '/home/alex/node_modules/deepspeech-tflite/index.d.ts'.
alex@portable-alex:~/codaz/Mozilla/DeepSpeech$ cat test.ts 
import DeepSpeech from "deepspeech-tflite";

let modelPath = '/path/to/model';
let model = new DeepSpeech.Model(modelPath);
model.setBeamWidth(1000);

let audioBuffer = [];
model.sttWithMetadata(audioBuffer);

DeepSpeech.FreeModel(model);
```

Reply:
Got it, LGTM. Glad to see it's working then.

Reply:
> Got it, LGTM. Glad to see it's working then.

If you say it's okay, I'll keep it then :-).